pages/en/cookbook/grafting.mdx

```diff
@@ -0,0 +1,215 @@
+---
+title: Replace a Contract and Keep its History With Grafting
+---
+
+In this guide, you will learn how to build and deploy new subgraphs by grafting existing subgraphs.
+
+## What is Grafting?
+
+Grafting reuses the data from an existing subgraph and starts indexing it at a later block. This is useful during development to get past simple errors in the mappings quickly or to temporarily get an existing subgraph working again after it has failed. Also, it can be used when adding a feature to a subgraph that takes long to index from scratch.
```

Comment:
The docs state:
> When a subgraph whose manifest contains a graft block is deployed, Graph Node will copy the data of the base subgraph up to and including the given block and then continue indexing the new subgraph from that block on. 

Does the base subgraph, onto which the one you currently running is grafted, have an end-block number? Intuitively I would say, it would have a start and end block.

Reply:
Yes, are you suggesting adding this info to the beginning?

Reply:
In a regular subgraph, you have startblock, and from that onwards, the contract is being indexed. A base subgraph though, should not be indexed anymore after a certain block number, right?
The docs state the following:
<img width="804" alt="image" src="https://user-images.githubusercontent.com/37071670/199353247-497726b5-4502-4955-8743-c606d72e87ca.png">
Which makes it seem that the base is going to be indexed indefinitely too.

Reply:
Don't think [endblock feature](https://github.com/graphprotocol/graph-node/issues/3504) exists yet, but thanks for the suggestions!